### ISSUE TITLE
 Error deleting Elastic IP association: InvalidAssociationID.NotFound

### ISSUE THREAD

**Issue by bizmate** [terraform-providers/terraform-provider-aws#1682]
### Terraform Version
$ terraform -v
Terraform v0.10.5

### Affected Resource(s)
- aws_eip_association 

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
Full gist - https://gist.github.com/bizmate/9bdf7de4a00ef2339807ec1ba2cc075d
Short story
```hcl
resource "aws_eip" "pivot_agent_public_ip" {
  #id                        = "eipalloc-57b6b864"
  vpc                       = true
  network_interface         = "${aws_network_interface.pivot_agent_network_interface.id}"
  associate_with_private_ip = "10.0.1.10"
  lifecycle {
    prevent_destroy = true
  }
}

resource "aws_eip_association" "eip_assoc" {
  instance_id   = "${aws_instance.pivot_gocd_agent.id}"
  allocation_id = "${aws_eip.pivot_agent_public_ip.id}"
}

resource "aws_instance" "pivot_gocd_agent" {
  ami           = "ami-1d4e7a66" # ubuntu 16.04 lts hvm:ebs-ssd
  instance_type = "t2.medium"

  tags {
    Name = "GoCD Agent for Pivot"
  }

  # The name of our SSH keypair we created above.
   key_name = "${aws_key_pair.auth.key_name}"

  # Our Security group to allow HTTP and SSH access
  vpc_security_group_ids = ["${aws_security_group.pivot_dev.id}"]

  # We're going to launch into the same subnet as our ELB. In a production
  # environment it's more common to have a separate private subnet for
  # backend instances.
  subnet_id = "${aws_subnet.subnet_pivot_dev.id}"

  #private_ip = "10.0.1.10"
  connection {
    # The default username for our AMI
    user = "ubuntu"
  }

  provisioner "remote-exec" {
    scripts = [
      "./bin/provision.sh"
    ]
  }

  provisioner "file" {
    source      = "config/gocd_agent.conf"
    destination = "/tmp/gocd_agent.conf"
  }

  provisioner "file" {
    source      = "config/autoregister.properties"
    destination = "/tmp/autoregister.properties"
  }

  # /var/lib/go-agent/config
  # /etc/default/go-agent

  provisioner "remote-exec" {
    scripts = [
      "./bin/start.sh"
    ]
  }

  lifecycle {
  	create_before_destroy = "true"
  }

}
```

### Debug Output
https://gist.github.com/bizmate/9bdf7de4a00ef2339807ec1ba2cc075d
see `destroy errors without finding association`

### Expected Behavior
Instance of ec2 and associated resources removed but IP

### Actual Behavior
somehow it cannot find association that was created minutes before

### Steps to Reproduce
1. `terraform apply` I create the infrastructure
2. `terraform destroy -force -target=aws_instance.pivot_gocd_agent` - I target the instance because i need to keep the IP reserved

### Important Factoids
EC2 classic

### References
unsure if linked to this https://github.com/hashicorp/terraform/issues/6758

**Comment by bizmate** [comment]
BTW i could not see this bug anymore. i m now using 010.7 and i have no clue if the provider version was updated and maybe fixed this

**Comment by bflad** [comment]
Bugfix PR submitted: #3653

**Comment by bflad** [comment]
The fix for this has been merged into master and will be released in v1.11.0 of the AWS provider, likely tomorrow. 👍

**Comment by bflad** [comment]
This has been released in [version 1.11.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1110-march-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
